core/src/main/scala/io/finch/streaming/StreamDecode.scala

```diff
@@ -0,0 +1,48 @@
+package io.finch.streaming
+
+import java.nio.charset.Charset
+
+import com.twitter.io.Buf
+import io.finch.Application
+import scala.annotation.implicitNotFound
+
+/**
+  * Stream HTTP streamed payload represented as S[F, Buf] into
+  * a S[F, A] of arbitrary type `A`.
+  */
+trait StreamDecode[S[_[_], _], F[_], A] {
```

Comment:
I believe this belongs in the top-level namespace.

Reply:
Also, by analogy with `DecodePath`, `DecodeEntity`, I'd probably vote naming this `DecodeStream`.